base/file.jl

```diff
@@ -420,19 +420,20 @@ const temp_prefix = "jl_"
 
 if Sys.iswindows()
 
-function _isroot(path::String)
+function ispathroot(path::String)
     m = match(r"^([^\\]+:|\\\\[^\\]+\\[^\\]+|\\\\\?\\UNC\\[^\\]+\\[^\\]+|\\\\\?\\[^\\]+:|)(.*)$", path)
-    m.captures[2] == "\\" ? true : false
+    (m.captures[2] == "\\" || isempty(m.captures[2])) ? true : false
```

Comment:
If `m.captures[2] == "\\"` then we don't need to add another `\`, right?    So we actually just want `isempty(m.captures[2])` for this case.    You also don't need the `... ? true : false`, since you can just return `...`.

Maybe it's better to just inline
```jl
# check whether path is the drive root or UNC path, missing the trailing backslash:
isroot = isempty(match(r"^([^\\]+:|\\\\[^\\]+\\[^\\]+|\\\\\?\\UNC\\[^\\]+\\[^\\]+|\\\\\?\\[^\\]+:|)(.*)$", path).captures[2])
```

Reply:
if `path = "C:"` then   `m.captures[2] = ""`   if `path = "C:\\"` then  `m.captures[2]== "\\"` . Path paths are roots.      Because `GetTempPathW` always adds a backslash `isempty(m.captures[2])`  is not needed, however I added it since the function tests if the path is rooted.

Reply:
I'm not sure how useful this function is in general, but it could be made to check is the path is a root for   non -windows OS,   by checking if the path starts with `/` ?